Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/manager_app/view/plan_list/PlanListActivity.java

```diff
@@ -121,7 +121,10 @@ private void setUpViews() {
 
     private void removePlan(long itemId){
         planTemplateRepository.delete(itemId);
-        searchView.setQuery(searchView.getQuery(), true);
+        if(searchView.getQuery().toString().isEmpty())
```

Comment:
Hi, thanks, it works now. As setQuery is not submitting the query if the string was empty string, it would be easier to read if we only write: 
```
planListAdapter.setPlanItems(planTemplateRepository.getFilteredByName(searchView.getQuery().toString()));
```
 We could avoid additional complication here. Please use it here, and extract it to method ->
``` 
refreshList(condition) 
```
that could be used in all 3 places ( lines 94, 100 )

Reply:
I am not sure if I understand, what would be the "condition" parameter of this function?

Reply:
function
```
private void refreshList(String searchedValue) {
   planListAdapter.setPlanItems(planTemplateRepository.getFilteredByName(searchedValue));
}
```

then use it 
``` 
refreshList(searchView.getQuery().toString()) 
```

Reply:
Oh, that's what you mean, okay. Fixed now

Reply:
Thanks. 👍